src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.cs

```diff
@@ -14,6 +18,70 @@ internal static void VerifyValidPath(string path, string argName)
             }
         }
 
+        internal static bool Copy(string sourcePath, string destinationPath, bool recursive,
+            bool skipExistingFiles, CancellationToken cancellationToken)
+        {
+            SearchOption searchOption = recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly;
+            IEnumerable<string> directoryEnumeration;
+            bool success = true;
+
+            try
+            {
+                EnumerationOptions enumerationOptions = EnumerationOptions.Compatible;
+
+                if (recursive)
+                {
+                    enumerationOptions = EnumerationOptions.CompatibleRecursive;
+                }
+
+                directoryEnumeration = Directory.EnumerateDirectories(sourcePath, "*", enumerationOptions);
+            }
+            catch (IOException)
+            {
+                // Return false if Enumeration is not possible
+                return false;
+            }
+
+            foreach (string enumeratedDirectory in directoryEnumeration)
+            {
+                string newDirectoryPath = enumeratedDirectory.Replace(sourcePath, destinationPath);
+                Directory.CreateDirectory(newDirectoryPath);
+
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            foreach (string enumeratedFile in Directory.GetFiles(sourcePath, "*.*", searchOption))
```

Comment:
Isn't `*.*` going to exclude files without extension?
```suggestion
            foreach (string enumeratedFile in Directory.GetFiles(sourcePath, "*", searchOption))
```